src/CascadeMinimizer.cc

```diff
@@ -286,7 +297,9 @@ bool CascadeMinimizer::hesse(int verbose ) {
       }
    }
 
+   freezeDiscParams(true);
    int iret = minimizer_->hesse(); 
```

Comment:
I remember hesse being quite fussy about running when the status of the floating/fixed parameters changes with respect to the last fit with migrad - did you check if this works ok if CascadeMinimizer::hesse() is called when MINIMIZER_analytic is not used (and the fit is not repeated in L286)?